src/api/controller/api/upload.js

```diff
@@ -138,37 +210,18 @@ export const checkChunkMiddleware = async ctx => {
     ctx.status = exists ? 200 : 204;
 };
 
-export const uploadUrl = async ctx => {
-    const { url, parserName } = ctx.request.body;
-    const [extension] = url.match(/[^.]*$/);
-
-    const parseStream = ctx.getParser(
-        !parserName || parserName === 'automatic' ? extension : parserName,
-    );
-
-    await ctx.dataset.remove({});
-
-    const stream = ctx.getStreamFromUrl(url);
-    const parsedStream = await parseStream(stream);
-
-    await ctx.saveStream(parsedStream);
-
-    ctx.status = 200;
-    ctx.body = {
-        totalLines: await ctx.dataset.count(),
-    };
-};
-
 const app = new Koa();
 
 app.use(prepareUpload);
 
 app.use(koaBodyParser());
 app.use(route.post('/url', uploadUrl));
+app.use(route.post('/url', saveParsedStream));
```

Comment:
Would better be used as a function call at the end of the `uploadUrl` route rather than as another middleware